### PR TITLE
Test suite work

### PR DESCRIPTION
**Pull request by Fuyukai**
Add underlying code allowing tests, add codecov coverage

### CODE REVIEW

**Comment by Fuyukai** [comment]
Don't care; didn't ask

**Comment by Fuyukai** [comment]
No idea why the Windows test is failing.

File: src/main/kotlin/util/CPythonCompiler.kt [hunk 1]
```diff
@@ -78,7 +81,18 @@ class CPythonCompiler() {
...
      */
     fun compile(data: String): KyCodeObject {
-        val args = listOf("--code", data)
-        return KyCodeObject(executeCompiler(args).code)
+        return if (!isWindows) {
+            val args = listOf("--code", data)
+            KyCodeObject(executeCompiler(args).code)
+        } else {
+            // windows requires a... different approach
+            val tmp = Files.createTempFile("kyc-windows-sucks", ".py")
```
Comment (ilevn) [inline src/main/kotlin/util/CPythonCompiler.kt:89]: Could be simplified with:
```kotlin
val tmp = createTempFile("kyc-windows-sucks", ".py").also {
    it.writeBytes(data.toByteArray(Charset.defaultCharset()))
}
val args = listOf("--path", tmp.absolutePath)
...
tmp.delete()
```
Windows, macOS and Linux also share UTF_8 as default charset, as far as I know.

**Review by ilevn — CHANGES_REQUESTED** [review]

File: src/main/kotlin/util/CPythonCompiler.kt (hunk 1 above)
Comment (ilevn) [reply src/main/kotlin/util/CPythonCompiler.kt:89]: Small addendum: If you don't need a ByteArray, `it.write(data, Charset.defaultCharset()` would also suffice

Comment (Fuyukai) [reply src/main/kotlin/util/CPythonCompiler.kt:89]: I don't see how that's simpler, this is pretty standard NIO API and it's a line/nest shorter

**Pull request merged**